certificate-renewal/REPWRITERSPECS/BANNO.CDRENEW.V1.POW

```diff
@@ -2318,10 +2367,117 @@ PROCEDURE EXPIREORDELETEXFERS
            SET EXPIRATIONDATE TO SYSTEMDATE
           END
         END
+       BNODEBUGMSG="SHARE XFER TYPE 2 DELETE/EXPIRE FMERROR='"+FMERROR+"'"
+       CALL BNODEBUGMSGADD
        TMPLOOP=TMPLOOP+1
       END
     END
   END
+ FOR TMPLOOP=1 TO STRANSFERCOUNT
+  DO
+   SXFERLOC(TMPLOOP)=0
+   SXFEREXPIREDATE(TMPLOOP)=DATENULL
+   SXFEREFFECTIVEDATE(TMPLOOP)=DATENULL
+  END
+END [PROCEDURE]
+
+PROCEDURE EXPIREORDELETETYPE3XFERS
+[* Build a list of current Share Transfer type 3 records and then,
+** depending upon the parameter setting, either delete or expire
+** the transfers.
+*]
+ STRANSFER3COUNT=0
+ FOR ACCOUNT TARGETACCOUNT
+  DO
+   FOR EACH SHARE WITH (SHARE:ID<>TARGETSHAREID AND
+                        SHARE:CLOSEDATE=DATENULL AND
+                        SHARE:CHARGEOFFDATE=DATENULL)
+    DO
+     BNODEBUGMSG="TESTING SHARE ID "+SHARE:ID+" FOR XFER 3 RECS"
+     CALL BNODEBUGMSGADD
+
+     FOR EACH SHARE TRANSFER WITH (SHARE TRANSFER:TYPE=3 AND
+                                   SHARE TRANSFER:ACCOUNTNUMBER=ACCOUNT:NUMBER AND
+                                   SHARE TRANSFER:IDTYPE=0 AND
+                                   SHARE TRANSFER:ID=TARGETSHAREID AND
+                                  (SHARE TRANSFER:EXPIRATIONDATE=DATENULL OR
+                                   SHARE TRANSFER:EXPIRATIONDATE>SYSTEMDATE) AND
+                                (((PARAMXFER3HANDLING=2 OR
+                                   PARAMXFER3HANDLING=3) AND
+                                   SHARE TRANSFER:NEXTDATE=SMATURITYDATE-1) OR
+                                   PARAMXFER3HANDLING>3))
```

Comment:
I think it would be more clear to explicitly check for `(PARAMXFER3HANDLING=4 OR PARAMXFER3HANDLING=5)`

Reply:
OK - done